packages/bibliography/src/main.rs

```diff
@@ -0,0 +1,358 @@
+    let mut database = Database::from_entries(bibliography.iter());
+
+    // What style to use (IEEE, APA etc)
+    let style_arg = input["arguments"]["style"].as_str().unwrap();
+
+    // Unwrap is safe since Core has checked enums already
+    let (bibliography_style, mut citation_style) = get_styles(style_arg).unwrap();
+
+    let mut output: Vec<Value> = vec![];
+
+    for citation_str in &citations {
+        // We parse the citation (which is a JSON obj)
+        let citation: InlineCitation = serde_json::from_str(citation_str).unwrap();
+
+        // We find the record by the key
+        let entry = if let Some(record) = database.records.get(citation.key) {
+            record.entry
+        } else {
+            eprintln!(
+                "Missing citation {0}, mentioned in [cite] {0} but not defined in the bibliography",
+                citation.key
+            );
+            output.push(add_citation_label!(
+                citation_str,
+                format!(
+                    "[{}]",
+                    text!(format!("[Missing citation '{}']", citation.key))
+                )
+            ));
+            continue;
+        };
+
+        // We construct a new citation (representing [cite] in Hayagravia)
+        let hayagriva_citation: Citation = Citation::new(entry, citation.note);
+
+        // We cite that entry from our db
+        let database_citation = database.citation(citation_style.as_mut(), &[hayagriva_citation]);
+
+        // From this, we construct our citation string (what the [cite] should be turned into).
+        // We pass this as JSON format to let Hayagravia apply formatting if needed (which it
+        // doesn't, in this version at least). We encase it in [] for IEE and () for others
+        let json_citation = if style_arg == "IEEE" {
+            let mut vec = vec![text!("[")];
+            vec.append(&mut display(&database_citation.display));
+            vec.push(text!("]"));
+            Value::Array(vec)
+        } else {
+            let mut vec = vec![text!("(")];
+            vec.append(&mut display(&database_citation.display));
+            vec.push(text!(")"));
+            Value::Array(vec)
+        };
+
+        // Then, we are adding that citation as the label (that the [cite], now [cite-internal],
+        // will turn into). Note, since we are doing string-matching later on (not value-based
+        // matching), we want to make sure not to parse citation_str and re-stringify it since
+        // object kv-pairs may change order. The string must match exactly to be picked up again
+        output.push(add_citation_label!(
+            citation_str,
+            format!("{json_citation}")
+        ));
+    }
+
+    let visibility = input["arguments"]["visibility"].as_str().unwrap();
+
+    // If the bibliography should be shown, show it!
+    if visibility == "shown" {
+        for entry in database.bibliography(bibliography_style.as_ref(), None) {
```

Comment:
I think the bibliography looks good but I think we should consider if we can style it some more (while keeping it flexible regarding the output format of course). For instance, the bracket with the numbers should end up in a column of the own instead of inside the text.

Also, it looks like it breaks on LaTeX output:
(But this looks to be my fault with the missing  `\n`  in the [newline] module that leads to TeX parsing the brackets as an argument to `\\` :sweat_smile: )
![image](https://user-images.githubusercontent.com/11508459/234206333-fb5999b5-32c9-4599-8799-33897f021496.png)

Reply:
I don't know why it doesn't render. I dont think it is an issue with `\\` since when adding a newline it still isn't shown. It works in math mode `$[2]$` but that seems unintended.

Reply:
Yeah, but the problem is that newlines are allowed to consume a pair of brackets as an option `\\[3cm]`

Reply:
Oh, didn't know about the newline option. Can we change newline to be `\\[]`?

Reply:
Or just `\\\n`

Reply:
It didn't work for me when adding a newline after `\\`. Does it for you? Otherwise, `\\[0pt]` works hehe

Reply:
`\\{}` works, that may be prefferable